src/Stratis.Bitcoin/Consensus/ChainedHeaderTree.cs

```diff
@@ -14,7 +14,22 @@ namespace Stratis.Bitcoin.Consensus
     /// </summary>
     public interface IChainedHeaderValidator
     {
-        void Validate(ChainedHeader chainedHeader);
+        /// <summary>
+        /// Validation of a header that was seen for the first time.
```

Comment:
The validator doesn't care whether it is first-time or not.

Reply:
Validates the block header.

Should probably be enough

Reply:
This is some extra useful information

Reply:
Yes but you would probably (and currently thats the case) always call this on new headers but not on headers there where already validated.

Its sort of similar to the comment bellow
> This validation represents minimal required validation for every block that we download.  
but you could call `VerifyBlockIntegrity` on blocks that where not downloaded (read form disk)